EnhancedMediaPlayer/AVPlayerView.swift

```diff
@@ -0,0 +1,69 @@
+// AVPlayerView.swift
+
+import AVKit
+import SwiftUI
+
+public struct AVPlayerView: UIViewControllerRepresentable {
+    private let playerState: MediaPlayerView.PlayerState
+    private let mediaURL: URL
+    private let onMediaFinished: () -> Void
+
+    init(
+        mediaURL: URL,
+        playerState: MediaPlayerView.PlayerState,
+        onMediaFinished: @escaping () -> Void
+    ) {
+        self.mediaURL = mediaURL
+        self.playerState = playerState
+        self.onMediaFinished = onMediaFinished
+    }
+
+    public func makeUIViewController(context: Context) -> AVPlayerViewController {
+        let playerController = AVPlayerViewController()
+        setupPlayerController(playerController)
+        updatePlayerState(playerController)
+        return playerController
+    }
+
+    public func updateUIViewController(_ playerController: AVPlayerViewController, context: Context) {
+        if let currentURL = playerController.player?.currentItem?.asset as? AVURLAsset,
+           currentURL.url != mediaURL {
+            setupPlayerController(playerController)
+        }
+        updatePlayerState(playerController)
+    }
+
+    private func setupPlayerController(_ playerController: AVPlayerViewController) {
+        playerController.showsPlaybackControls = false
+        playerController.player = AVPlayer(url: mediaURL)
+        playerController.observeMediaFinished(onEnd: onMediaFinished)
+    }
+    
+    private func updatePlayerState(_ playerController: AVPlayerViewController) {
+        switch playerState {
+            case .playing: playerController.player?.play()
+            default: playerController.player?.pause()
+        }
+    }
+}
+
+extension AVPlayerViewController {
+    func observeMediaFinished(onEnd: @escaping () -> Void) {
+        NotificationCenter.default.addObserver(
+            forName: NSNotification.Name.AVPlayerItemDidPlayToEndTime,
+            object: self.player?.currentItem,
+            queue: nil
+        ) { _ in
+            onEnd()
+            self.removeObserver()
+        }
+    }
+
+    private func removeObserver() {
+        NotificationCenter.default.removeObserver(
+            self,
+            name: .AVPlayerItemDidPlayToEndTime,
+            object: nil
+        )
+    }
```

Comment:
I think it makes more sense to subscribe to these events in the `MediaPlayerController`, don't you? Instead of passing a `mediaURL: URL` to `AVPlayerView`, you can pass an instance of an `AVPlayer` directly. Instantiate the instance (and keep its reference) in the `MediaPlayerController`. With that, you'll be able to, in a private method, do this:
```swift
NotificationCenter.default.addObserver(
    forName: .AVPlayerItemDidPlayToEndTime,
    object: self.avPlayer.currentItem,
    queue: DispatchQueue.main
) {
    // whatever 
}
```